ydb/core/tx/schemeshard/ut_login/ut_login.cpp

```diff
@@ -177,6 +192,171 @@ Y_UNIT_TEST_SUITE(TSchemeShardLoginTest) {
         resultLogin = Login(runtime, "user12", "passwordu12*#");
         UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "");
     }
+
+    Y_UNIT_TEST(AccountLockoutAndAutomaticallyUnlock) {
+        TTestBasicRuntime runtime;
+        runtime.AddAppDataInit([](ui32 nodeIdx, NKikimr::TAppData& appData) {
+            Y_UNUSED(nodeIdx);
+            auto accountLockout = appData.AuthConfig.MutableAccountLockout();
+            // AttemptThreshold = 4
+            accountLockout->SetAttemptResetDuration("5s");
+        });
+        TTestEnv env(runtime);
+        ui64 txId = 100;
+        CreateAlterLoginCreateUser(runtime, ++txId, "/MyRoot", "user1", "password1");
+        auto resultLogin = Login(runtime, "user1", "wrongpassword1");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "Invalid password");
+        resultLogin = Login(runtime, "user1", "wrongpassword2");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "Invalid password");
+        resultLogin = Login(runtime, "user1", "wrongpassword3");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "Invalid password");
+        resultLogin = Login(runtime, "user1", "wrongpassword4");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "Invalid password");
+        resultLogin = Login(runtime, "user1", "wrongpassword5");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), TStringBuilder() << "User user1 is locked out");
+
+        runtime.SimulateSleep(TDuration::Seconds(7));
+
+        resultLogin = Login(runtime, "user1", "wrongpassword6");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "Invalid password");
+        resultLogin = Login(runtime, "user1", "password1");
+        UNIT_ASSERT_VALUES_EQUAL(resultLogin.error(), "");
+        auto describe = DescribePath(runtime, TTestTxConfig::SchemeShard, "/MyRoot");
```

Comment:
This code is repeated in other tests. Let's move it to the function.

Reply:
done